[FIX] compare the sum of all quantity by product in lines between tranfer and invoice TASK#4914 HU#959 CA#2

hugho-ad opened the pull request:
**[WIP]**
- [x] Verify qty and product bewtween each transfer & invoice lines

FIX https://github.com/Vauxoo/yoytec/issues/500

**DUMMY https://github.com/Vauxoo/yoytec/pull/1068**

hugho-ad committed:
[FIX] compare the sum of all quantity by product in lines between tranfer and invoice

[FIX] compare the sum of all quantity by product in lines between tranfer and invoice

hugho-ad commented:
@JulioSerna 
Can you review please.

I'm open to include any verifications you considers necessary

Regards

JulioSerna commented on stock_picking_invoice_validation/model/stock_transfer_details.py:
```diff
@@ -51,25 +51,27 @@ def do_detailed_transfer(self):
+                        StockTansferDetails, self).do_detailed_transfer()
+                    return res
                 else:
                     raise exceptions.Warning(
```
@hugho-ad 
what happen if is partial delivery?,
e.g invoice line 10 quantity transfer item 1 quantity

hugho-ad replied:
That transfer process will be done until quantities of each product in picking out, are completely availables in order to complete delivery based on invoice lines.

In other words, if a picking out are incompleate, we can't transfer to the customer.

I requested more details in HU 959, Until the response, the above are my thought.

Thanks for you support.
Regards!

hugho-ad committed:
[FIX] count invoce lines with product type consu or product

hugho-ad commented:
@JulioSerna can you help me to merge this please.

nhomar commented on stock_picking_invoice_validation/model/sale_order.py:
```diff
@@ -8,17 +8,24 @@
+        check_inv_pick = self.env["ir.config_parameter"].get_param(
+            key)
+        return check_inv_pick
+
```
Hello.

Functional checks like this one can not be in ir.config.parameter, they must have its entries in config setting in the warehous section, please fix that.

hugho-ad committed:
[REF] Change field type from Selection Field to Boolean Field. Is more faster compare Boolean values than strings

nhomar commented:
@hugho-ad I was waiting this module has a config no a ir.config.parameter.
 Show all checks

hugho-ad commented:
@nhomar itś has in warehouse section:
![image](https://cloud.githubusercontent.com/assets/10233975/14859903/d0a00ca0-0c6b-11e6-92db-737a2f3c1616.png)

---

nhomar commented:
@hugho-ad I mean that the parameter must be saved in other place (human readable) not in the ir.config-parameter object.

the company is a good place.

nhomar commented on stock_picking_invoice_validation/model/sale_order.py:
```diff
@@ -8,26 +8,27 @@
+    def _get_wh_conf(self):
+        key = "stock.check_inv_pick"
+        check_inv_pick = self.env["ir.config_parameter"].get_param(key)
+        return check_inv_pick
```
read this from the company.

hugho-ad committed:
[REF] Change the main configuration.

Change the main configuration of checking invoices vs pickings from
stock.config.settings to res.company model

 # Please enter the commit message for your changes. Lines starting

hugho-ad committed:
[REF] update translation files

[REF] update translation files

nhomar commented:
Is this ready @hugho-ad , rebase please.

hugho-ad commented:
rebase Done, merge it please

hugho-ad commented:
@nhomar Rebase Done again, 
After mege this
Mege this :https://github.com/Vauxoo/yoytec/pull/1067 (TODO: Remove dummy commit)

nhomar commented on stock_picking_invoice_validation/model/stock_transfer_details.py:
```diff
@@ -50,26 +46,29 @@ def do_detailed_transfer(self):
+                    res = super(
+                        StockTansferDetails, self).do_detailed_transfer()
+                    return res
                 else:
```
Please No el... return early.

nhomar commented on stock_picking_invoice_validation/model/stock_transfer_details.py:
```diff
@@ -50,26 +46,29 @@ def do_detailed_transfer(self):
+                          'between moves and invoice lines.'))
+                    # raise exceptions.Warning(
+                    #     _('Warning!'), _('Incorrect Invoice'))
             else:
```
Again... No else... return early.

nhomar commented:
Minimal comments in algorithm.. avoid the else returning earliest as commented.

hugho-ad commented:
@nhomar comments done, and CI green 

Dummy : https://github.com/Vauxoo/yoytec/pull/1067
